libs/tup-components/src/tickets/TicketModal.tsx

```diff
@@ -0,0 +1,379 @@
+              prepareRow(row);
+              return (
+                <tr {...row.getRowProps()}>
+                  {row.cells.map((cell) => (
+                    <td {...cell.getCellProps()}>{cell.render('Cell')}</td>
+                  ))}
+                </tr>
+              );
+            })
+          ) : (
+            <tr>
+              <td colSpan={5}>No attachments to display.</td>
+            </tr>
+          )}
+        </tbody>
+      </table>
+    </div>
+  );
+};
+Attachments.propTypes = {
+  attachments: PropTypes.arrayOf(PropTypes.array.isRequired).isRequired,
+  ticketId: PropTypes.string.isRequired,
+};
+
+const TicketHistoryReply: React.FC<{ ticketId: string }> = ({ ticketId }) => {
+  const mutation = useTicketReply(ticketId);
+  const { mutate, isLoading, isError } = mutation;
+
+  const defaultValues = useMemo(
+    () => ({
+      text: '',
+      files: [],
+    }),
+    []
+  );
+
+  return (
+    <Formik
+      enableReinitialize
+      initialValues={defaultValues}
+      validationSchema={formSchema}
+      onSubmit={(values, { resetForm }) => {
+        const formData = new FormData();
+        formData.append('text', values['text']);
+        if (values.files) {
+          values.files.forEach((file) => formData.append('files', file));
+        }
+        mutate(formData, {
+          onSuccess: () => resetForm(),
+        });
+      }}
+    >
+      {({ isSubmitting, isValid }) => {
+        return (
+          <Form className="ticket-reply-form">
+            <FormikInput
+              name="text"
+              label="Reply"
+              type="textarea"
+              className="ticket-reply-text-area"
+              description=""
+              required
+            />
+            <FileInputDropZoneFormField
+              id="files"
+              isSubmitted={isSubmitting}
+              description="Error reports and screenshots can be helpful for diagnostics"
+              maxSizeMessage="Max File Size: 3MB"
+              maxSize={3145728}
+            />
+            <FormGroup className="ticket-reply-submission">
+              {isError && (
+                <InlineMessage type="error">
+                  Something went wrong.
+                </InlineMessage>
+              )}
+              <Button
+                attr="submit"
+                type="primary"
+                disabled={!isValid || isSubmitting || isLoading || isError}
+                isLoading={isLoading}
+              >
+                Reply
+              </Button>
+            </FormGroup>
+          </Form>
+        );
+      }}
+    </Formik>
+  );
+};
+
+TicketHistoryReply.propTypes = {
+  ticketId: PropTypes.string.isRequired,
+};
+
+const TicketHistoryCard: React.FC<historyCardParams> = ({
+  created,
+  creator,
+  isCreator,
+  content,
+  attachments,
+  ticketId,
+}) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleIcon = isOpen ? (
+    <Icon name="icon-action icon-collapse" />
```

Comment:
**Icon Name is Different Than We Think:**

```suggestion
    <Icon name="icon-action icon-contract" />
```

Also, I'll add an alias to @tacc/core-styles.

Reply:
Self, you added an alias in [TACC/Core-Styles#85](https://github.com/TACC/Core-Styles/pull/85). It is not available in tup-ui yet, but will be eventually.